lib/io/hardware_uart.hpp

```diff
@@ -69,8 +69,8 @@ class Uart : public IUart {
   Uart(core::ILogger &logger, const int file_descriptor);
   ~Uart();
 
-  core::Result sendBytes(const char *tx, const std::uint8_t length);
-  core::Result readBytes(unsigned char *rx, const std::uint8_t length);
+  virtual core::Result sendBytes(const char *tx, const std::uint8_t length);
+  virtual core::Result readBytes(unsigned char *rx, const std::uint8_t length);
```

Comment:
Should these be `virtual ..... () = 0;`? Or am I forgetting where that should go

Reply:
that syntax defines pure virtual functions, i.e. ones that must be derived from, but these are not

Reply:
Oh yea me dumb